crypto/rand/build.info

```diff
@@ -1,7 +1,19 @@
 LIBS=../../libcrypto
 
-$COMMON=rand_lib.c rand_crng_test.c rand_win.c rand_unix.c  rand_vms.c \
-        drbg_lib.c drbg_ctr.c rand_vxworks.c drbg_hash.c drbg_hmac.c
+$COMMON=rand_lib.c rand_crng_test.c rand_unix.c \
+        drbg_lib.c drbg_ctr.c drbg_hash.c drbg_hmac.c
+IF[{- !$disabled{'egd'} -}]
+  $COMMON=$COMMON rand_egd.c
+ENDIF
+IF[{- $config{target} =~ /vxworks/i -}]
+  $COMMON=$COMMON rand_vxworks.c
+ENDIF
+IF[{- $config{target} =~ /vms/i -}]
+  $COMMON=$COMMON rand_vms.c
+ENDIF
+IF[{- $target{perl_platform} =~ /windows/i -}]
+  $COMMON=$COMMON rand_win.c
+ENDIF
```

Comment:
If this bit proves difficult, I would undo it for now.  I think that windows in particular may be tricky, because of things like Cygwin and Mingw...